cartridge/upload.lua

```diff
@@ -0,0 +1,313 @@
+--- Spread the data across instances in a network-efficient manner.
+--
+-- (**Added** in v2.4.0-32)
+--
+-- @module cartridge.upload
+-- @local
+
+local log = require('log')
+local fio = require('fio')
+local errno = require('errno')
+local fiber = require('fiber')
+local checks = require('checks')
+local errors = require('errors')
+local digest = require('digest')
+local msgpack = require('msgpack')
+local pool = require('cartridge.pool')
+local utils = require('cartridge.utils')
+
+local vars = require('cartridge.vars').new('cartridge.upload')
+
+local UploadError = errors.new_class('UploadError')
+
+--- The uploaded data.
+-- `{[upload_id] = data}`
+-- @table inbox
+vars:new('inbox', {})
```

Comment:
It could be managed externally. Maybe as a Tarantool future.
`upload` could result some object that has some condition. We could wait result, interrupt uploading, etc via such object.
Also it should be user responsibility how to manage result of upload. And it solves a problem when in case of failure we store garbage on filesystem - `__gc` metamethod or `cleanup` callback should remove this garbage.

Reply:
It's a distributed system. The uploaded data can't be deleted before the sender makes another netbox call (`prepare_2pc`). The only thing I could do is a time-managed GC, but it's not a cure - overcomplicated and there still may be problems.